hearth/lib/hearth/context.rb

```diff
@@ -13,17 +13,16 @@ def initialize(options = {})
       @params = options[:params]
       @metadata = options[:metadata] || {}
       @interceptors = options[:interceptors] || InterceptorList.new
-      @interceptor_attributes = {}
     end
 
     # @return [Symbol] Name of the API operation called.
     attr_reader :operation_name
 
     # @return [Hearth::HTTP::Request]
-    attr_reader :request
+    attr_accessor :request
 
     # @return [Hearth::HTTP::Response]
-    attr_reader :response
+    attr_accessor :response
```

Comment:
This can still be a reader right? since we are just calling `reset` no the existing one.

Reply:
For stub response, we should make it an accessor. I think it makes sense to replace in that case, so that the test has access to response for assertions.

Reply:
I'm not sure I understand, stub responses can modify the request, which allows tests to have access to responses for assertions.

And if the response is replaced, any wrappers put on the response body (eg, a verify checksum) will be lost.

Reply:
Fair point. I think both need to be readers now, and I'll bring back the replace method. Since we are modifying the request (reset signing) instead of dup.